File: src/main/java/net/gegy1000/terrarium/server/map/cover/generator/layer/SelectionSeedLayer.java
package net.gegy1000.terrarium.server.map.cover.generator.layer;

import net.minecraft.world.gen.layer.GenLayer;
import net.minecraft.world.gen.layer.IntCache;

public class SelectionSeedLayer extends GenLayer {
    private final int range;

    public SelectionSeedLayer(int range, long seed) {
        super(seed);
        this.range = range;
    }

    @Override
    public int[] getInts(int areaX, int areaY, int areaWidth, int areaHeight) {
        int[] result = IntCache.getIntCache(areaWidth * areaHeight);
        for (int z = 0; z < areaHeight; z++) {
            for (int x = 0; x < areaWidth; x++) {
                this.initChunkSeed(areaX + x, areaY + z);
                result[x + z * areaWidth] = this.nextInt(this.range);
            }
        }
        return result;
    }
}

File: src/main/kotlin/net/gegy1000/terrarium/server/map/glob/generator/layer/SelectionSeedLayer.kt
package net.gegy1000.terrarium.server.map.glob.generator.layer

import net.minecraft.world.gen.layer.GenLayer
import net.minecraft.world.gen.layer.IntCache

class SelectionSeedLayer(val range: Int, seed: Long) : GenLayer(seed) {
    override fun getInts(areaX: Int, areaY: Int, areaWidth: Int, areaHeight: Int): IntArray {
        val result = IntCache.getIntCache(areaWidth * areaHeight)
        for (z in 0..areaHeight - 1) {
            for (x in 0..areaWidth - 1) {
                this.initChunkSeed((areaX + x).toLong(), (areaY + z).toLong())
                result[x + z * areaWidth] = this.nextInt(this.range)
            }
        }
        return result
    }
}
